File: Leet_Code-1189/Leet_Code-1189.py
'''
Question 1189
'''

class Solution:
    def maxNumberOfBalloons(self, text: str) -> int:
        b = text.count('b')
        a = text.count('a')
        l = text.count('l')//2
        o = text.count('o')//2
        n = text.count('n')
        if b == 0 or a == 0 or l == 0 or o == 0 or n == 0:
            return 0
        return min(b,a,l,o,n)

File: Leet_Code-1189/Leet_Code-1189.java
/*
 * Question 1189
 */

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

class Solution {
    public int maxNumberOfBalloons(String text) {
        List<Character> textList = new ArrayList<>();
        for (char c : text.toCharArray()) {
            textList.add(c);
        }
        List<Integer> counts = new ArrayList<>();
        counts.add(Collections.frequency(textList, 'b'));
        counts.add(Collections.frequency(textList, 'a'));
        counts.add(Collections.frequency(textList, 'l')/2);
        counts.add(Collections.frequency(textList, 'o')/2);
        counts.add(Collections.frequency(textList, 'n'));

        if (counts.contains(0)){
            return 0;
        }
        return Collections.min(counts);
    }
}